src/main/cpp/src/get_json_object.cu

```diff
@@ -80,106 +60,39 @@ thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
   return thrust::nullopt;
 }
 
-/**
- * TODO: JSON generator
- *
- */
-template <int max_json_nesting_depth = curr_max_json_nesting_depth>
-class json_generator {
- public:
-  CUDF_HOST_DEVICE json_generator(char* _output, size_t _output_len)
-    : output(_output), output_len(_output_len)
-  {
-  }
-  CUDF_HOST_DEVICE json_generator() : output(nullptr), output_len(0) {}
-
-  // create a nested child generator based on this parent generator
-  // child generator is a view
-  CUDF_HOST_DEVICE json_generator new_child_generator()
-  {
-    if (nullptr == output) {
-      return json_generator();
-    } else {
-      return json_generator(output + output_len, 0);
-    }
-  }
-
-  CUDF_HOST_DEVICE json_generator finish_child_generator(json_generator const& child_generator)
-  {
-    // logically delete child generator
-    output_len += child_generator.get_output_len();
-  }
-
-  CUDF_HOST_DEVICE void write_start_array()
-  {
-    // TODO
-  }
-
-  CUDF_HOST_DEVICE void write_end_array()
-  {
-    // TODO
-  }
-
-  CUDF_HOST_DEVICE void copy_current_structure(json_parser<max_json_nesting_depth>& parser)
-  {
-    // TODO
-  }
-
-  /**
-   * Get current text from JSON parser and then write the text
-   * Note: Because JSON strings contains '\' to do escape,
-   * JSON parser should do unescape to remove '\' and JSON parser
-   * then can not return a pointer and length pair (char *, len),
-   * For number token, JSON parser can return a pair (char *, len)
-   */
-  CUDF_HOST_DEVICE void write_raw(json_parser<max_json_nesting_depth>& parser)
-  {
-    if (output) {
-      auto copied = parser.write_unescaped_text(output + output_len);
-      output_len += copied;
-    }
-  }
-
-  CUDF_HOST_DEVICE inline size_t get_output_len() const { return output_len; }
-
- private:
-  char const* const output;
-  size_t output_len;
-};
-
-/**
- * @brief Result of calling a parse function.
- *
- * The primary use of this is to distinguish between "success" and
- * "success but no data" return cases.  For example, if you are reading the
- * values of an array you might call a parse function in a while loop. You
- * would want to continue doing this until you either encounter an error
- * (parse_result::ERROR) or you get nothing back (parse_result::EMPTY)
- */
-enum class parse_result {
-  ERROR,          // failure
-  SUCCESS,        // success
-  MISSING_FIELD,  // success, but the field is missing
-  EMPTY,          // success, but no data
-};
+template <int max_json_nesting_depth>
+CUDF_HOST_DEVICE bool evaluate_path(json_parser<max_json_nesting_depth>& p,
```

Comment:
Nit: Typically we should not use host+device qualifier too much. Only use it for short computation functions. If it does not need to be on device then just declare as a normal function.

Reply:
Refer to this [comment](https://github.com/NVIDIA/spark-rapids-jni/pull/1868#discussion_r1530015452)